server/utils/utils.ts

```diff
@@ -166,3 +166,168 @@ export function getComponentString(macroName: string, params = {}): string {
 
   return nunjucks.renderString(macroString, {})
 }
+
+/**
+ * Adds missing question mark to a question.
+ *
+ * If a sentence is a question (based on the start of it) and it's missing
+ * the question marks it adds it. The input string is trimmed.
+ *
+ * Examples:
+ * - `'   DO BIRDS FLY  '` => `'DO BIRDS FLY?'`
+ * - `'When did this happen?'` => `'When did this happen?'` (unchanged, already ended with question mark)
+ * - `'Describe the incident'` => `'Describe the incident'` (unchanged, not a question)
+ * correctly (i.e. each part in a double-barreled is converted to proper case).
+ *
+ * @param label sentence to adjust
+ * @returns trimmed sented with a question mark if it's a question.
+ */
+export function addQuestionMarkToQuestion(label: string): string {
+  let result = label.trim()
+
+  const questionsPrefixes = [
+    'are',
+    'can',
+    'did',
+    'do',
+    'does',
+    'from what',
+    'from which',
+    'has',
+    'have',
+    'how',
+    'is',
+    'to which',
+    'was',
+    'were',
+    'what',
+    'when',
+    'where',
+    'which',
+    'who',
+    'whose',
+    'why',
+    'with what',
+  ]
+  const labelStart = result.toLowerCase()
+  if (questionsPrefixes.some(prefix => labelStart.startsWith(`${prefix} `))) {
+    if (!result.endsWith('?')) {
+      result += '?'
+    }
+  }
+
+  return result
+}
+
+/**
+ * Converts a string to sentence case
+ *
+ * First letter upper case, the rest in lower case. Preserve original
+ * case of acronyms and other special words.
+ *
+ * @param str input string
+ * @returns string in sentence case
+ */
+export function convertToSentenceCase(str: string): string {
+  // Special words (e.g. acronyms) - Preserve case if possible
+  const preserveList = [
+    'A1',
+    'ACCT',
+    'ADA',
+    'AFFF',
+    'AirPods',
+    'Amazon',
+    'Apple',
+    'BDH',
+    'C.I',
+    'C.N',
+    'C.S',
+    'CANDR',
+    'CCTV',
+    'CO2',
+    'CPS',
+    'CRC',
+    'DIMU',
+    'DJI',
+    'DMIU',
+    'DVD',
+    'DYI',
+    'EGS',
+    'F2052SH',
+    'F78A',
+    'FES',
+    'GP',
+    'H.C.C.',
+    'HCO',
+    'HHMD',
+    'HMPS',
+    'HSE',
+    'IEP',
+    'IMB',
+    'iPad',
+    'KPI',
+    'Lewis',
+    'LSD',
+    'MiFi',
+    'N/A',
+    'NDTSG',
+    'NOMS',
+    'NOU',
+    'NPS',
+    'NTRG',
+    'O.C',
+    'OMA',
+    'P.E.',
+    'PEMS',
+    'PPCS',
+    'ROTL',
+    'SFO',
+    'SIM',
+    'SSU',
+    'Stockholm',
+    'T/R',
+    'Tornado',
+    'TV',
+    'UAL',
+    'UAV',
+    'UKBA',
+    'USB',
+    'UV',
+    'VPU',
+    'YO',
+    // 'IT', // IT is problematic: IT (Information Technology) or "it" pronoun?
+  ]
+
+  const input = str.trim()
+  if (input.length === 0) {
+    return ''
+  }
+
+  // match words or non-words
+  const regex = /(\w+|[^\w\s]+|\s)/g
+  const parts = input.match(regex)
```

Comment:
`parts` can be null if for some reason the input does not match (e.g. on empty string), in which case i would simply return the input unchanged

Reply:
`input` is trimmed and returned as-is if it's an empty string (well, if it's length is `0`) - so it shouldn't be an issue? 🤔 

Beside mapped would then be an empty array and `.join('')` would turn into an empty string anyway no?